content/ui-patterns/loading.mdx

```diff
@@ -0,0 +1,406 @@
+
+## Lifecycle of a loading state
+
+<img
+  width="960"
+  alt="4 images: 1. explicitly excluded loading indicator; 2. button in loading state; 3. success banner; 4. error banner"
+  src="https://github.com/primer/design/assets/2313998/91f1ecd4-78e3-4178-a505-749ab05f3efa"
+/>
+
+**Process initiated:** The moment just before loading begins. Usually initiated by a user action such as clicking a button or submitting a form
+
+**Process in progress:** The process is underway. This is the actual loading state.
+
+**Process complete successfully:** The process has finished successfully, and the user can continue with their task. If the user isn't navigated to a new context when the task has completed, this step may include some indication of success, such as a green checkmark or a success message.
+
+**Process failed:** There was an error, and the process could not be completed. This step should include an error message that explains what went wrong and what the user's next time may be. For example, if the user can retry the process, include a button to do so.
+
+## Incremental loading
+
+Whenever possible, show each item in a collection as soon as it loads. Don't wait for every item in the collection to load before showing the items.
+
+<DoDontContainer>
+  <Do>
+    <CustomVideoPlayer
+      width="456"
+      loop
+      src="https://github.com/primer/design/assets/2313998/6e7f87df-0d3e-4a58-a001-dd09e4510cc3"
+    />
+    <Caption>Show content as soon as it's loaded.</Caption>
+  </Do>
+  <Dont>
+    <CustomVideoPlayer
+      width="456"
+      loop
+      src="https://github.com/primer/design/assets/2313998/a06a4ee2-cbe6-4ec0-8551-16381c7bee49"
+    />
+    <Caption>Don't wait for all content in a collection to be loaded before showing it.</Caption>
+  </Dont>
+</DoDontContainer>
+
+Whenever possible, prioritize the most important pieces of data to load first. This shows the user the most important content first and gives them a chance to interact with things sooner. This could be especially helpful for somebody on a slow network.
+
+<img
+  width="960"
+  alt="4 images of a repo card from the feed loading: 1. Skeleton; 2. Text; 3. Actions; 4. Images"
+  src="https://github.com/primer/design/assets/2313998/ea5725c7-bec2-43a0-8cd9-43fba1a27a51"
+/>
+
+## Scoping loading indicators to page content
+
+A loading indicator should be placed nearest to the content they are standing in for. Avoid creating a jarring layout shift when the loaded content replaces the loading indicator.
+
+Replace loading indicators with loaded content or error feedback as soon as possible to avoid burdening users with unnecessarily long wait times.
+
+### Interstitial loading screen
+
+<CustomVideoPlayer
```

Comment:
This video actually shows a scenario I was thinking we might want to cover. There have been a few accessibility violations about loading buttons used within a dialog on submit, where submitting the form loads a new page. In most cases we actually just removed the loading button since the new page loading indicates a state change. 

So not really related to this content section, but wanted to point it out. Although the video is of production so I don't expect you to change it 😆 also, the staff bar is visible in this video FYI.

Reply:
Haha oops! Let me re-record without the staff bar 😅